station-items-v0/src/main/java/net/modificationstation/stationapi/mixin/item/MixinPlayerBase.java

```diff
@@ -26,4 +27,15 @@ private void hijackInteractWith(EntityBase arg, CallbackInfo ci) {
         if (getHeldItem() != null && getHeldItem().getType() instanceof UseOnEntityFirst use && use.onUseOnEntityFirst(getHeldItem(), (PlayerBase) (Object) this, this.level, arg))
             ci.cancel();
     }
+
+
+    @Inject(method = "attack", at = @At("HEAD"), cancellable = true)
+    public void attack_preHit(EntityBase entity, CallbackInfo ci){
+        ItemInstance itemInstance = this.getHeldItem();
+        if(itemInstance != null){
+            if(!itemInstance.preHit((Living) entity, PlayerBase.class.cast(this))){
```

Comment:
Causes a crash if the entity hit isn't `Living`. Either allow non-living entities into the method arguments, or add an `instanceof` check to the previous if statement.

I think allowing non-living entities into method arguments is better, since the damage logic applies to all entities anyway and not just `Living`, and someone might want to cancel that using `preHit`.
Though it will make arguments different to vanilla's `postHit`, the return type is different anyway, so I think changing method arguments is acceptable.

Reply:
Allowed any entity as an argument